pkg/kubelet/cm/node_container_manager_linux.go

```diff
@@ -53,6 +53,33 @@ func (cm *containerManagerImpl) createNodeAllocatableCgroups() error {
 	return nil
 }
 
+func (cm *containerManagerImpl) updateInternalCapacity(rl v1.ResourceList) bool {
+	for k, v := range rl {
+		if _, ok := cm.internalCapacity[k]; !ok {
+			continue
+		}
+
+		if !v.Equal(cm.internalCapacity[k]) {
+			klog.Infof("node allocatable resource %q has changed", k)
+			cm.internalCapacity[k] = v
+			return true
+		}
+	}
+
+	return false
+}
+
+// UpdateNodeAllocatableCgroups updates cgroups for node allocatable resources.
+func (cm *containerManagerImpl) UpdateNodeAllocatableCgroups(rl v1.ResourceList) error {
+
+	if cm.updateInternalCapacity(rl) {
+		klog.Error("Failed to enforce cgroups")
+		return cm.enforceNodeAllocatableCgroups()
```

Comment:
Not sure if

```go

if err := cm.enforceNodeAllocatableCgroups(); err != nil {
     return fmt.Errorf("Failed to enforce cgroups: %+v", err)
}
```
is prefered instead of just logging. Will defer that to @mattjmcnaughton :smile: